ApiBundle/Transformer/NodeGroupRoleTransformer.php

```diff
@@ -52,25 +66,43 @@ public function transform($nodeGroupRole)
     }
 
     /**
-     * @param FacadeInterface             $facade
+     * @param GroupInterface              $group
+     * @param FacadeInterface             $nodeRoleFacade
      * @param NodeGroupRoleInterface|null $source
      *
      * @throws RoleNotFoundHttpException
-     * @return mixed
+     * @throws TransformerParameterTypeException
+     *
+     * @return null|NodeGroupRoleInterface
      */
-    public function reverseTransform(FacadeInterface $facade, $source = null)
+    public function reverseTransformWithContext(GroupInterface $group, FacadeInterface $nodeRoleFacade, $source = null)
```

Comment:
I would call the method `reverseTransformWithGroup` to show that you have to give a group and this way you could differentiate all the following method names